packages/roles/tests/php/test_Roles.php

```diff
@@ -0,0 +1,128 @@
+	 */
+	public function test_current_user_to_role_with_role() {
+		$this->mock_function( 'current_user_can', true, 'administrator' );
+		
+		$this->assertEquals( 'administrator', $this->roles->translate_current_user_to_role() );
+	}
+
+	/**
+	 * @covers Automattic\Jetpack\Roles::translate_current_user_to_role
+	 */
+	public function test_current_user_to_role_with_capability() {
+		$this->mock_function( 'current_user_can', true, 'edit_others_posts' );
+		
+		$this->assertEquals( 'editor', $this->roles->translate_current_user_to_role() );
+	}
+
+	/**
+	 * @covers Automattic\Jetpack\Roles::translate_current_user_to_role
+	 */
+	public function test_current_user_to_role_with_no_match() {
+		$this->mock_function( 'current_user_can', false );
+		
+		$this->assertEquals( false, $this->roles->translate_current_user_to_role() );
+	}
+
+	/**
+	 * @covers Automattic\Jetpack\Roles::translate_user_to_role
+	 */
+	public function test_user_to_role_with_role() {
+		$user_mock = $this->getMockBuilder( 'WP_User' )->getMock();
+		$this->mock_function( 'user_can', true, $user_mock, 'administrator' );
+		
+		$this->assertEquals( 'administrator', $this->roles->translate_user_to_role( $user_mock ) );
+	}
+
+	/**
+	 * @covers Automattic\Jetpack\Roles::translate_user_to_role
+	 */
+	public function test_user_to_role_with_capability() {
+		$user_mock = $this->getMockBuilder( 'WP_User' )->getMock();
+		$this->mock_function( 'user_can', true, $user_mock, 'edit_others_posts' );
+		
+		$this->assertEquals( 'editor', $this->roles->translate_user_to_role( $user_mock ) );
+	}
+
+	/**
+	 * @covers Automattic\Jetpack\Roles::translate_user_to_role
+	 */
+	public function test_user_to_role_with_no_match() {
+		$user_mock = $this->getMockBuilder( 'WP_User' )->getMock();
+		$this->mock_function( 'user_can', false );
+		
+		$this->assertEquals( false, $this->roles->translate_user_to_role( $user_mock ) );
+	}
+
+	/**
+	 * @covers Automattic\Jetpack\Roles::translate_role_to_cap
+	 */
+	public function test_role_to_cap_existing_role() {
+		$this->assertEquals( 'edit_others_posts', $this->roles->translate_role_to_cap( 'editor' ) );
+	}
+
+	/**
+	 * @covers Automattic\Jetpack\Roles::translate_role_to_cap
+	 */
+	public function test_role_to_cap_non_existing_role() {
+		$this->assertEquals( false, $this->roles->translate_role_to_cap( 'follower' ) );
+	}
+
+	/**
+	 * Mock a global function and make it return a certain value.
+	 * Optionally can limit the mock to invocations with certain arguments.
+	 *
+	 * @param string $function_name Name of the function.
+	 * @param mixed  $return_value  Return value of the function.
+	 * @param mixed  $arg_1_value   Value of the first argument value we expect.
+	 * @param mixed  $arg_2_value   Value of the second argument value we expect.
+	 * @return phpmock\Mock The mock object.
+	 */
+	protected function mock_function( $function_name, $return_value = null, $arg_1_value = null, $arg_2_value = null ) {
```

Comment:
at some point we should create a package with all this test helpers

Reply:
Yeah, good point, and I agree.

The reason why I haven't gone that far to do it yet is because I want to see some more patterns arising, and then we can pick the best ones of them.

I haven't written thousands of packages tests yet, but I already see some differences in patterns between test helpers in packages, so let's wait and see what's best, and we can wrap stuff up as a package.